django/thunderstore/api/utils.py

```diff
@@ -0,0 +1,67 @@
+from django.conf import settings
+from drf_yasg.utils import swagger_auto_schema, unset
+
+ENABLED_API_DOCS_CLASSES = [
+    "DeprecateModApiView.post",
+    "PackageViewSet.list",
+    "PackageViewSet.retrieve",
+    "PackageViewSet.rate",
+    "CurrentUserInfoView.get",
+    "CommunitiesExperimentalApiView.get",
+    "PackageCategoriesExperimentalApiView.get",
+    "CurrentCommunityExperimentalApiView.get",
+    "PackageListingUpdateApiView.post",
+    "CommunityPackageListApiView.get",
+    "FrontPageApiView.get",
+    "RenderMarkdownApiView.post",
+    "PackageDetailApiView.get",
+    "LegacyProfileCreateApiView.post",
+    "LegacyProfileRetrieveApiView.get",
+    "PackageListApiView.get",
+    "PackageDetailApiView.get",
+    "PackageVersionDetailMixin.get",
+    "PackageVersionDetailApiView.get",
+    "PackageVersionChangelogApiView.get",
+    "PackageVersionReadmeApiView.get",
+    "UploadPackageApiView.get",
+    "UploadPackageApiView.post",
+    "SubmitPackageApiView.post",
+    "ReadmeValidatorApiView.post",
+    "ManifestV1ValidatorApiView.post",
+    "IconValidatorApiView.post",
+    "PackageWikiApiView.get",
+    "PackageWikiApiView.post",
+    "PackageWikiApiView.delete",
+    "PackageWikiListAPIView.get",
+    "CompleteLoginApiView.post",
+    "CurrentUserExperimentalApiView.get",
+    "DeleteSessionApiView.post",
+    "OverwolfLoginApiView.post",
+    "ValidateSessionApiView.get",
+    "SchemaChannelApiView.post",
+    "SchemaChannelLatestApiView.get",
+    "UserMediaInitiateUploadApiView.post",
+    "UserMediaFinishUploadApiView.post",
+    "UserMediaAbortUploadApiView.post",
+    "WikiPageApiView.get",
+]
+
+if settings.SHOW_QA_API_ENDPOINTS_IN_DOCS:
+    ENABLED_API_DOCS_CLASSES = ENABLED_API_DOCS_CLASSES + [
+        "CommunityListAPIView.get",
+        "TeamDetailAPIView.get",
+        "CommunityDetailAPIView.get",
+    ]
```

Comment:
You don't need to maintain this list at all. If something isn't conditionally rendered, just use the default decorator. If it is conditionally rendered, then pass the condition in the decorator as a parameter instead of comparing names.